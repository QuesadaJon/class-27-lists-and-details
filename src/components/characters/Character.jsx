import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, affiliation, photoUrl }) => (
  <figure>
    <figcaption>{name}</figcaption>
    <img src={photoUrl}/>
    <p>{affiliation}</p>
  </figure>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired,
  photoUrl: PropTypes.string.isRequired
};

export default Character;
