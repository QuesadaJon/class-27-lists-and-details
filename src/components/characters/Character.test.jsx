import React from 'react';
import { render } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
  it('matches a snapshot', () => {
    const { asFragment } = render(
      <Character 
        name="Samuel"
        photoUrl="http//image.com"
        affiliation="affiliation"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
