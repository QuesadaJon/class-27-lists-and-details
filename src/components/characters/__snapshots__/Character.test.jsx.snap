// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component matches a snapshot 1`] = `
<DocumentFragment>
  <figure>
    <figcaption>
      Samuel
    </figcaption>
    <img
      src="http//image.com"
    />
    <p>
      affiliation
    </p>
  </figure>
</DocumentFragment>
`;
